src/Halogen/HTML.purs

```diff
@@ -96,6 +98,15 @@ slot_
 slot_ label p component input =
   Core.widget (ComponentSlot (componentSlot label p component input (const Nothing)))
 
+-- | Renders `ComponentHTML` to an arbitrary element in the DOM instead of its
+-- | actual location within a component's render tree.
+portal
+  :: forall action slots m
+   . HTMLElement
+  -> ComponentHTML action slots m
+  -> ComponentHTML action slots m
+portal elem html = Core.widget (PortalSlot elem html)
```

Comment:
Another thing I considered was a type signature like this:

```purs
portal
  :: forall action slots m
   . Maybe HTMLElement
  -> ComponentHTML action slots m
  -> ComponentHTML action slots m
```

where a `Nothing` would mean to use the `<body>` tag, which from my experience is by far the most common case, and a `Just htmlElement` means to use the provided element. Curious what y'all think.

Reply:
Or, alternately, another function:

```
portal_
  :: forall action slots m
   . ComponentHTML action slots m
  -> ComponentHTML action slots m
portal_ = portalImpl Nothing

-- Allows you to create a portal to a target element
portal
  :: forall action slots m
   . HTMLElement
  -> ComponentHTML action slots m
  -> ComponentHTML action slots m
portal elem = portalImpl (Just elem) 

-- Purely internal, shared by portal_ and portal but not exported
portalImpl
  :: forall action slots m
   . Maybe HTMLElement
  -> ComponentHTML action slots m
  -> ComponentHTML action slots m
```